### PR TITLE
Fix compatibility with --incompatible_objc_provider_remove_linking_info

### PR DESCRIPTION
**Pull request by keith**
Fixes https://github.com/bazelbuild/rules_apple/issues/2353

### CODE REVIEW

**Comment by keith** [comment]
depends on https://github.com/bazelbuild/rules_swift/pull/1143

File: apple/internal/framework_import_support.bzl [hunk 1]
```diff
@@ -457,8 +460,30 @@ def _objc_provider_with_dependencies(
...
 
+def _new_dynamic_framework_provider(*, cc_info, **kwargs):
+    """A wrapper API for the Bazel API of the same name to better support multiple Bazel versions
+
+    Args:
+        cc_info: CcInfo provider for the dynamic framework.
+        **kwargs: Additional arguments to pass if supported.
+    """
+
+    test_provider = apple_common.new_dynamic_framework_provider(
```
Comment (brentleyjones) [inline apple/internal/framework_import_support.bzl:476]: Same as my comment in rules_swift, can we avoid creating a throw-away instance?

Comment (keith) [reply apple/internal/framework_import_support.bzl]: done

File: apple/internal/linking_support.bzl [hunk 2]
```diff
@@ -51,6 +54,26 @@ def _debug_outputs_by_architecture(link_outputs):
...
+def _new_executable_binary_provider(*, binary, cc_info, objc):
+    """Wrap the apple_common API of the same name to better support multiple Bazel versions.
+
+    Args:
+        binary: The `File` representing the binary.
+        cc_info: The `CcInfo` provider for the binary.
+        objc: The `apple_common.Objc` provider for the binary.
+    """
+    if _OBJC_PROVIDER_LINKING:
+        return apple_common.new_executable_binary_provider(
```
Comment (keith) [inline apple/internal/linking_support.bzl:66]: it looks to me like these rules are the only consumers of this info, so theoretically we could switch to our own provider for this instead, if we did the only field we currently use is the `binary`

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**